Source/CarthageKit/Project.swift

```diff
@@ -745,10 +771,13 @@ public final class Project { // swiftlint:disable:this type_body_length
 			}
 
 		return loadResolvedCartfile()
-			.flatMap(.merge) { resolvedCartfile in
-				return self
-					.buildOrderForResolvedCartfile(resolvedCartfile, dependenciesToInclude: dependenciesToCheckout)
-					.collect()
+			.map { resolvedCartfile -> [(Dependency, PinnedVersion)] in
+				let dependencies = resolvedCartfile.dependencies
+				return dependencies.keys
+					.filter { dependenciesToCheckout?.contains($0.name) ?? true }
+					.map { key in
+						(key, dependencies[key]!)
```

Comment:
Well dependencies is a dictionary, and we want to process each item, but that can be worked down a level when it's zipped with the submodule dictionary, rather than doing it here to force it into an array.